src/ui/dice.test.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2018 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import React from 'react';
+import { Dice } from './dice';
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+Enzyme.configure({ adapter: new Adapter() });
+
+test('basic', () => {
+  {
+    const dice = Enzyme.shallow(<Dice />);
+    expect(dice.html()).toContain('div');
+  }
+  {
+    const dice = Enzyme.shallow(<Dice className="custom" />);
+    expect(dice.html()).toContain('custom');
+  }
+  {
+    const dice = Enzyme.shallow(<Dice dotStyle={false} faceValue={1} />); // renders plain text instead of styled dots
+    expect(dice.text()).toContain(1);
+  }
+});
+
+test('state change', () => {
+  {
+    const dice = Enzyme.mount(<Dice faceValue={4} />);
+    dice.setProps({ faceValue: 1 });
+    expect(dice.state().faceValue).toBe(1);
+    dice.setProps({ faceValue: 1 });
+    expect(dice.state().faceValue).toBe(1);
+  }
+});
+
+test('onClick', () => {
+  {
+    const dice = Enzyme.mount(<Dice faceValue={4} />);
+    expect(dice.state().faceValue).toBe(4);
+  }
+  {
+    const onClick = jest.fn();
+    const dice = Enzyme.mount(<Dice onClick={onClick} faceValue={4} />);
+    dice.simulate('click');
+    expect(onClick).toHaveBeenCalled();
+    expect(dice.state().faceValue).not.toBe(0);
+  }
+  {
+    const onClick = jest.fn();
```

Comment:
you may create 
```js
describe('onClick, ...
let onClick;
beforeEach(()=>{
 onClick = jest.fn()
}
test('',...

```

Reply:
when you use tree hierarchy of tests you can describe what cases you testing and it the good way for cut duplication of code

Reply:
I'm not sure in this case it helps too much. Additionally, it follows the pattern in the rest of the testing files so that would be part of a larger refactoring.